<a href="https://colab.research.google.com/github/RiteshopShrivastava/Assignment-3-Hypothesis-Testing/blob/main/Assignment_3_Q4_Hypothesis_Testing.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

Q 4. TeleCall uses 4 centers around the globe to process customer order forms. They audit a certain % of the customer order forms. Any error in order form renders it defective and has to be reworked before processing. The manager wants to check whether the defective % varies by centre. Please analyze the data at 5% significance level and help the manager draw appropriate inferences

DATASETS: CustomerOrderForm.mtw

In [13]:
import pandas as pd
import numpy as np
from scipy import stats
from scipy.stats import chi2_contingency

In [6]:
customer = pd.read_csv('/content/Costomer+OrderForm.csv')
customer

Phillippines   Indonesia       Malta       India
0     Error Free  Error Free   Defective  Error Free
1     Error Free  Error Free  Error Free   Defective
2     Error Free   Defective   Defective  Error Free
3     Error Free  Error Free  Error Free  Error Free
4     Error Free  Error Free   Defective  Error Free
..           ...         ...         ...         ...
295   Error Free  Error Free  Error Free  Error Free
296   Error Free  Error Free  Error Free  Error Free
297   Error Free  Error Free   Defective  Error Free
298   Error Free  Error Free  Error Free  Error Free
299   Error Free   Defective   Defective  Error Free

[300 rows x 4 columns]

# Chi square Test
Ho: The defective % of customer order form does not varies by centre

H1 : The defective %  of customer order form varies by centre

In [7]:
customer.Phillippines.value_counts()

Error Free    271
Defective      29
Name: Phillippines, dtype: int64

In [8]:
customer.Indonesia.value_counts()

Error Free    267
Defective      33
Name: Indonesia, dtype: int64

In [9]:
customer.Malta.value_counts()

Error Free    269
Defective      31
Name: Malta, dtype: int64

In [10]:
customer.India.value_counts()

Error Free    280
Defective      20
Name: India, dtype: int64

In [12]:
array = np.array([[271,267,269,280],[29,33,31,20]])
array

array([[271, 267, 269, 280],
       [ 29,  33,  31,  20]])

In [14]:
chi2_contingency(array) # returns chi square stats, p, Dof, observed val

(3.858960685820355,
 0.2771020991233135,
 3,
 array([[271.75, 271.75, 271.75, 271.75],
        [ 28.25,  28.25,  28.25,  28.25]]))

Since (pvalue=0.277) > (alpha=0.05), therfore accept null hypothesis.

***Inference : Accept Ho :The defective % of customer order form does not varies by centre***